packages/roosterjs-editor-plugins/lib/plugins/CustomReplace/CustomReplace.ts

```diff
@@ -99,6 +103,7 @@ export default class CustomReplacePlugin implements EditorPlugin {
 
         const searcher = this.editor.getContentSearcherOfCursor(event);
         const stringToSearch = searcher.getSubStringBefore(this.longestReplacementLength);
+        const sourceTag = this.editor.getElementAtCursor().tagName;
```

Comment:
Same question, how to handle `<sub><span>****</span></sub>`

Reply:
It does not, it only gets the first html tag the cursor is in. This function is intended for replacing a string to another string while someone is typing. If someone inserts their own html CustomReplace does not make replacements.

Reply:
I think what Andres is trying to do is get the tag of cursor position, then compare if the tag html that is being in inserted, then the should replace will do the logic to replace or not

Reply:
It is just one more information to use in shouldReplace callback. Is that it @Andres-CT98?

Reply:
My point is, we should not rely on the direct element tag. If you want to check whether user is inputting under a special tag, you need to find it all the way above until the root of editor. For example:
```html
const sub = editor.getElementAtCursor('sub');
if (sub) {
 ... 
}
```
In that way, no matter what is inside SUB, it can always find it.

Reply:
And always remember user can input content inside the content that is pasted from other places. So we should never assume something can't exist in the content

Reply:
I understand users can insert anything, but replacement rules only apply while typing.

Reply:
User can paste something, then put cursor inside the pasted content, then type. In that case, we still need to understand what kind of HTML structure it is.

Reply:
That is not the task of this function, that is for whatever replacement rules users write to define the behaviour. Passing the editor will give them more tools though, so I think it is the optimal implementation.

Reply:
No matter what function it is, we should never rely on the tag name of current direct container element since HTML structure can be variant. User can simply press Ctrl+B to add a `<b>` tag around, then what you get will be "B" rather than the one you want to check.

Reply:
And as a publish API, once we add a new parameter, we can't change or remove it until we bump the major version otherwise it will become a breaking change. So that we need to be super careful when we add new parameter to public API, and try to make it general. If we found tagName is not enough someday and need to pass in more info, how can we do it? Then we still need to add editor parameter at that time. Then it will result in two parameter: tagName and editor. But which one should we use in a callback? What if they are conflict? All these makes it into a bad design.

Reply:
Now I understand, the user can use CustomReplace inside a pasted content, then the tag name is not reliable anymore.